### DISCUSSION TITLE
Getting backend errors from Slurm

### DISCUSSION THREAD

**Discussion by grishchalgeri2014** [cesar-douady/open-lmake#21] [General]
lmake -v  -J output2.txt
lmakeserver: fatal: No hash plugins loaded. Was slurm_init() called before calling any Slurm API functions?

I am seeing backend errors for slurm. Slurm is up and running correctly.

xyz@ubuntubuild:~/examples$ sinfo
PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
debug*       up   infinite      1   idle localhost
xyz@ubuntubuild:~/examples$ ls -ltr /opt/slurm/lib/libslurm.so
lrwxrwxrwx 1 root root 18 Aug  9 19:04 /opt/slurm/lib/libslurm.so -> libslurm.so.42.0.0

echo $LD_LIBRARY_PATH 
/opt/slurm/lib:/opt/slurm/lib/slurm:/opt/open-lmake-0.1/lib:/opt/slurm/lib/slurm:/opt/slurm/lib:

echo $PATH
/opt/slurm/bin:/opt/slurm/sbin:/opt/open-lmake-0.1/bin/:/usr/bin/python3.8:/home/pchalger/open-lmake/py_env/bin:/opt/slurm/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

The slurm seems running and initialized correctly.

Below is my Lmake python file

import sys
import os
import socket

sys.path.append('/opt/open-lmake-0.1/lib')

import lmake
from lmake.rules import Rule

# Check if Slurm backend is available
if 'slurm' in lmake.backends:

    # Configure Slurm backend
    lmake.config.backends.slurm = {'interface': socket.gethostname()}

    # Manually define the sources to avoid automatic source detection issues
    lmake.manifest = ['Lmakefile.py', 'job1.sh', 'job2.sh']

    # Rule to submit a job to Slurm
    class Job1(Rule):
        stems = {'File': '.*'}
        targets = {'DST': 'output1.txt'}
        cmd = './job1.sh'
        backend = 'slurm'
        resources = {'mem': '20M'}

    # Rule to submit another job to Slurm
    class Job2(Rule):
        stems = {'File': '.*'}
        targets = {'DST': 'output2.txt'}
        deps = {'SRC': 'output1.txt'}
        cmd = './job2.sh'
        backend = 'slurm'
        resources = {'mem': '10M'}

else:
    print("Slurm backend is not implemented.")

**Comment by cesar-douady** [comment]
Hi grish-stack,
I was actually not calling slurm_init during initialization.
I am surprised that this had no impact till now.
This is fixed in both branches v24.05 and advanced.

**Comment by cesar-douady** [comment]
By the way, did the unit tests included in the code base (automatically run when you type make) pass ?
They shouldn't have. In particular "make unit_tests/slurm.dir/tok" should have failed.

**Comment by grishchalgeri2014** [comment]
it did fail for me but the actual examples succeeded.

**Reply by cesar-douady** [reply]
Did the fix work ?

**Comment by grishchalgeri2014** [comment]
Just tried it and the fix did work for me.
